department-of-veterans-affairs/caseflow
# Display Hearings Scheduled column in the landing page

Resolves https://vajira.max.gov/browse/CASEFLOW-1042

### Description
- batch query vacols and postress to determine the slot for each hearing day
- feature toggle the display and download of this column - [slack convo](https://dsva.slack.com/archives/C01MBDKAC7P/p1625582263061300?thread_ts=1625089585.054000&cid=C01MBDKAC7P)

### Acceptance Criteria (Reviewers should check these off)
- [x] Hearings schedule includes a new "Hearings Scheduled" column displaying number of hearings scheduled and total number of slots for each hearing day (e.g., "5 of 8")
- [x] When user clicks "Download current view", the column is also included in the downloaded CSV
- [x] 🧘‍♂️ Figure out why user timing event isn't sent to Google Analytics that measures load time of hearing days in the list
Looks like we're using ApiUtil, and I'd expect this code to send a timing event for the POST: https://github.com/department-of-veterans-affairs/caseflow/blob/ed0a3fcf3f565167329be37d7652046f5465ccbd/client/app/util/ApiUtil.js#L16

### Testing Plan
1. Login as `BVASYELLOW` and to go the http://localhost:3000/hearings/schedule. Ensure the new column does **not** display on page and does **not**  appear on the CSV
2. on Console, run `FeatureToggle.enable!(:view_and_download_hearing_scheduled_column)`
1.  Login as `BVASYELLOW` and to go the http://localhost:3000/hearings/schedule. Ensure the new column **does** on page and does appear on the CSV

### User Facing Changes
 BEFORE|AFTER
 ---|---
<img width="1301" alt="Screen Shot 2021-07-06 at 12 04 24 PM" src="https://user-images.githubusercontent.com/20735998/124634376-48818900-de54-11eb-8791-a9141576971e.png">|<img width="1212" alt="Screen Shot 2021-07-06 at 12 03 38 PM" src="https://user-images.githubusercontent.com/20735998/124634408-4fa89700-de54-11eb-96d3-5ea05951d8be.png">

### Release Plans
1. Turn Feature toggle on for Dave
3. Deploy this branch to PROD
